### PR TITLE
Add new error message when class implements class

### LINKED ISSUE #19793: Improve error message when `implements` is used instead of `extends`
**TypeScript Version:**  2.7.0-dev.201xxxxx

**Code**

```ts
class A {
	private x: number;
}

class B implements A {}
```

**Expected behavior:**

Begin the error message with `class 'B' is declared to implement class 'A'. Did you mean 'extends'?`. Then continue with the specific error message.

**Actual behavior:**

```
src/a.ts(5,7): error TS2420: Class 'B' incorrectly implements interface 'A'.
  Property 'x' is missing in type 'B'.
```
Not helpful on its own because we can't implement `x`. I figure 99% of the time one intends to extend a class rather than implement, so we should just show an additional message any time one class implements another and there's an error.

### PR DESCRIPTION
**Pull request by charlespierce**
Fixes #19793 

Added new diagnostic message for when a class incorrectly implements another class. Also added that diagnostic to the `errorCodes` list for the `fixClassIncorrectlyImplementsInterface` CodeFix

### CODE REVIEW

File: src/compiler/diagnosticMessages.json [hunk 1]
```diff
@@ -2228,6 +2228,10 @@
         "category": "Error",
         "code": 2716
     },
+    "Class '{0}' is declared to implement class '{1}'. Did you mean 'extends'?": {
```
Comment (mhegazy) [inline src/compiler/diagnosticMessages.json]: i would rather this stayed `class '{0}' incorrectly implements class '{1}'. Did you mean 'extends'?`

Comment (mhegazy) [reply src/compiler/diagnosticMessages.json]: @DanielRosenwasser thoughts about this one?

Comment (charlespierce) [reply src/compiler/diagnosticMessages.json]: I can make a quick change to that effect, I was just using the error message from the issue description.

Comment (charlespierce) [reply src/compiler/diagnosticMessages.json]: Updated.

**Comment by charlespierce** [comment]
PR Updated to resolve merge conflicts.

File: src/compiler/diagnosticMessages.json [hunk 2]
```diff
@@ -2260,6 +2260,10 @@
         "category": "Error",
         "code": 2718
     },
+    "Class '{0}' incorrectly implements class '{1}'. Did you mean 'extends'?": {
```
Comment (DanielRosenwasser) [inline src/compiler/diagnosticMessages.json]: I'd prefer if the phrasing was more a little more specific on what's being discussed. Maybe something more like:

```
Class '{0}' incorrectly implements class '{1}'. Did you mean to extend '{1}' and inherit its members as a subclass?
```

**Comment by charlespierce** [comment]
@DanielRosenwasser @mhegazy Any remaining concerns with this PR?

**Comment by charlespierce** [comment]
@mhegazy @DanielRosenwasser Updated to resolve conflicts in `diagnosticMessages.json`

**Pull request merged**